### ISSUE TITLE
Errors in Frequency Format 

### ISSUE THREAD

**Issue by michaelamoako** [fairlearn/fairlearn#821]
Rather than Error Rates, it would be helpful to have a generalizable way of showing errors in Frequency format. 

#### Is your feature request related to a problem? Please describe.

Mental processing of Fairlearn output. 

For example, if FPR is 5E-5, that is hard to understand. If instead it were "Approximately __ errors out of every ___ occurences", that would be more understandable 

#### Describe the solution you'd like

Implementation of Frequency format for error metrics

#### Describe alternatives you've considered, if relevant

Application of function that takes error rates, and turns them into frequency format .

#### Additional context

The frequency format hypothesis is the idea that the brain understands and processes information better when presented in frequency formats rather than a numerical or probability format. See [(Frequency Format Hypothesis)](https://en.wikipedia.org/wiki/Frequency_format_hypothesis#:~:text=The%20frequency%20format%20hypothesis%20is,20%25%20leads%20to%20better%20comprehension.)

**Comment by michaelamoako** [comment]
@riedgar-ms any workarounds?

**Comment by riedgar-ms** [comment]
This sounds more like a job for the final printout, rather than `MetricFrame`. We could probably write a formatter to do something like this, though. Figuring out a useful set of heuristics for the descriptions might be non-trivial, though.

**Comment by hildeweerts** [comment]
I agree with @riedgar-ms it might be tricky implement this generally. But it might make sense to do this for our predefined base metrics (e.g., `true_positive_rate`) and predefined fairness metrics (e.g., `equalized_odds_difference`). wdyt @michaelamoako?

**Comment by riedgar-ms** [comment]
@hildeweerts this isn't really about metrics per se. This is how, given a small number, to render it in a "x times in y" way which preserves some level of accuracy. but there are all sorts of questions to answer. These centre around specifying that accuracy and perception of the numbers. For example, is the distinction between "5 times in 127" and "9 times in 253" meaningful to people? Those are obviously pretty close, but you have to stop and match denominators to see that.

**Comment by romanlutz** [comment]
I've encountered the same thing with the chest x-ray classification case study and my workaround was to always calculate:
- metrics of interest
- count (merged in today with #812 but can be manually implemented with 1 line as well)
- # of positives
- % of positives

You can do the same for error rate! 
- # of errors
- % of errors / error rate

They should appear next to each other in the `by_group` table and you can interpret it much easier.

Reducing it to the lowest common denominator may be more of a challenge, but I honestly don't quite see the point in doing that. For all numbers we should have a way of formatting the number output if someone doesn't like `5E-5`, for example. I can't quite say how one could do that but I'm sure pandas provides that as a configuration option somewhere.

**Comment by romanlutz** [comment]
Can anyone see a reasonable way forward in this issue? 

The much more narrow issue of how to format metrics to avoid `5E-5` and instead have `.000` (restricted to some number of decimals) remains, of course. Or at least I'm not aware of any options we provide for people right now to configure that.

**Comment by riedgar-ms** [comment]
I maintain that this is a text-formatter issue, not Fairlearn per se - and a potential nightmare of edge cases.

**Comment by hildeweerts** [comment]
A quick google search revealed quite a few options to change the display format in pandas: https://re-thought.com/how-to-suppress-scientific-notation-in-pandas/

We could add this as an example to the user guide, but I agree that it is probably more related to pandas than Fairlearn.

**Comment by riedgar-ms** [comment]
The `%` format option could help as well, by lopping off some of the leading zeros.
